### PR TITLE
Merge Loan Split and Loan tabs

### PR DESCRIPTION
**Pull request by kenneth-fung**
I tried to make this addition as unobtrusive as possible. No real change should be noticed when setting a command's category and when making new panels and tabs for the UI.

The UI now checks if a `CommandCategory` is a "secondary" category. If it is, the contents of a tab will be swapped from its `primaryPanel` to its `secondaryPanel`. Otherwise, the `primaryPanel` will be used. Technically, we could make it such that each tab has multiple panels to choose from, but since I don't think such a scenario will come up, I left it such that each tab has just one `secondaryPanel`.

### CODE REVIEW

File: src/main/java/budgetbuddy/ui/MainWindow.java [hunk 1]
```diff
@@ -156,11 +155,23 @@ void fillInnerParts() {
...
-            pane.getSelectionModel().select(tabToView);
+    private void updateView(CommandCategory category) {
+        requireNonNull(category);
+
+        PanelTab tabToView = tabMap.get(
+                category.isSecondaryCategory()
+                        ? category.getPrimaryCategory(category)
+                        : category);
+
+        requireNonNull(tabToView);
```
Comment (tcgj) [inline src/main/java/budgetbuddy/ui/MainWindow.java]: I don't think you should requireNonNull here. Commands like `help` or `exit` will have exceptions thrown in that case. Those commands should have categories but no tabs. Only update the current tab if the tabToView isn't null

File: src/main/java/budgetbuddy/ui/tab/PanelTab.java [hunk 2]
```diff
@@ -32,4 +39,18 @@ public PanelTab(DisplayPanel panel, String imageFileName) {
             }
         });
     }
+
+    public PanelTab(DisplayPanel primaryPanel, DisplayPanel secondaryPanel, String imageFileName) {
+        this(primaryPanel, imageFileName);
+        this.secondaryPanel = secondaryPanel;
+    }
```
Comment (tcgj) [inline src/main/java/budgetbuddy/ui/tab/PanelTab.java]: This looks a little iffy, perhaps set the other constructor to call this one instead?

File: src/main/java/budgetbuddy/logic/commands/CommandCategory.java [hunk 3]
```diff
@@ -3,6 +3,19 @@
...
+    public boolean isSecondaryCategory() {
+        return this == LOAN_SPLIT;
+    }
+
+    public CommandCategory getPrimaryCategory(CommandCategory secondary) {
+        if (secondary == LOAN_SPLIT) {
+            return LOAN;
+        }
+        return this;
+    }
```
Comment (tcgj) [inline src/main/java/budgetbuddy/logic/commands/CommandCategory.java]: Wow hmm this seems a little bit hardcoded ish

Comment (kenneth-fung) [reply src/main/java/budgetbuddy/logic/commands/CommandCategory.java]: I agree, but figured that:

- If someone else wants to add another secondary category, they can add on to the checks in these methods.

- No one else is likely to use it anyway.

The main problem I had to solve was to signal to the UI that a category was not going to the match the tab name. I’m open to suggestions on how better to accomplish this, although this seems like the easiest way.

File: src/main/java/budgetbuddy/ui/MainWindow.java (hunk 1 above)
Comment (kenneth-fung) [reply src/main/java/budgetbuddy/ui/MainWindow.java]: That’s a good point, I’ll move the checking of category back down and change the parameter here back to the tab to view.

File: src/main/java/budgetbuddy/ui/tab/PanelTab.java (hunk 2 above)
Comment (kenneth-fung) [reply src/main/java/budgetbuddy/ui/tab/PanelTab.java]: Sorry, I’m not sure how that might work. How would the other constructor know whether to call this second one? It doesn’t have the secondary panel passed in.

Comment (tcgj) [reply src/main/java/budgetbuddy/ui/tab/PanelTab.java]: So the other constructor will only have `this(panel, null, imageFileName);` wheres this panel will set all 3 instance variables and also `setClosable(false)` etc.

File: src/main/java/budgetbuddy/logic/commands/CommandCategory.java (hunk 3 above)
Comment (tcgj) [reply src/main/java/budgetbuddy/logic/commands/CommandCategory.java]: fair enough!

File: src/main/java/budgetbuddy/ui/tab/PanelTab.java (hunk 2 above)
Comment (kenneth-fung) [reply src/main/java/budgetbuddy/ui/tab/PanelTab.java]: Ohh, so all the setting of variables are done in one constructor? Makes sense, I'll amend the code.

**Pull request merged**